src/application.coffee

```diff
@@ -354,33 +350,30 @@ apiPollInterval = (val) ->
 	clearInterval(updateStatus.intervalHandle)
 	application.poll()
 
-specialActionEnvVars =
-	'RESIN_OVERRIDE_LOCK': null # This one is in use, so we keep backwards comp.
-	'RESIN_SUPERVISOR_DELTA': null
-	'RESIN_SUPERVISOR_UPDATE_STRATEGY': null
-	'RESIN_SUPERVISOR_HANDOVER_TIMEOUT': null
-	'RESIN_SUPERVISOR_OVERRIDE_LOCK': null
+specialActionConfigVars =
 	'RESIN_SUPERVISOR_VPN_CONTROL': utils.vpnControl
 	'RESIN_SUPERVISOR_CONNECTIVITY_CHECK': utils.enableConnectivityCheck
 	'RESIN_SUPERVISOR_POLL_INTERVAL': apiPollInterval
 	'RESIN_SUPERVISOR_LOG_CONTROL': utils.resinLogControl
 
-executedSpecialActionEnvVars = {}
+executedSpecialActionConfigVars = {}
 
-executeSpecialActionsAndHostConfig = (env) ->
+executeSpecialActionsAndHostConfig = (conf, oldConf) ->
 	Promise.try ->
-		_.map specialActionEnvVars, (specialActionCallback, key) ->
-			if env[key]? && specialActionCallback?
+		_.map specialActionConfigVars, (specialActionCallback, key) ->
+			if conf[key]? && specialActionCallback?
 				# This makes the Special Action Envs only trigger their functions once.
-				if !_.has(executedSpecialActionEnvVars, key) or executedSpecialActionEnvVars[key] != env[key]
-					logSpecialAction(key, env[key])
-					specialActionCallback(env[key])
-					executedSpecialActionEnvVars[key] = env[key]
-					logSpecialAction(key, env[key], true)
-		hostConfigVars = _.pick env, (val, key) ->
-			return _.startsWith(key, device.hostConfigEnvVarPrefix)
-		if !_.isEmpty(hostConfigVars)
-			device.setHostConfig(hostConfigVars, logSystemMessage)
+				if !_.has(executedSpecialActionConfigVars, key) or executedSpecialActionConfigVars[key] != conf[key]
+					logSpecialAction(key, conf[key])
+					specialActionCallback(conf[key])
+					executedSpecialActionConfigVars[key] = conf[key]
+					logSpecialAction(key, conf[key], true)
+		hostConfigVars = _.pick conf, (val, key) ->
+			return _.startsWith(key, device.hostConfigConfigVarPrefix)
+		oldHostConfigVars = _.pick oldConf, (val, key) ->
+			return _.startsWith(key, device.hostConfigConfigVarPrefix)
+		if !_.isEmpty(hostConfigVars) && !_.isEqual(hostConfigVars, oldHostConfigVars)
```

Comment:
This should drop the `!_.isEmpty(hostConfigVars)` check, so that's it's possible to remove all of the host config vars that might have been set